tests/test_conformer.py

```diff
@@ -84,3 +90,234 @@ def get_output(shape, norm):
+        print(sequence_mask.shape)
+
+        cfg = VGG4LayerActFrontendV1Config(
+            conv1_channels=conv1_channels,
+            conv2_channels=conv2_channels,
+            conv3_channels=conv3_channels,
+            conv4_channels=conv4_channels,
+            conv_kernel_size=(3, 3),
+            conv_padding=None,
+            pool1_kernel_size=pool1_red,
+            pool1_stride=pool1_red,
+            pool1_padding=None,
+            pool2_kernel_size=pool2_red,
+            pool2_stride=pool2_red,
+            pool2_padding=None,
+            activation=nn.functional.relu,
+            linear_input_dim=in_dim,
+            linear_output_dim=out_dim,
+        )
+
+        output, sequence_mask = VGG4LayerActFrontendV1(cfg)(data_input, sequence_mask)
+
+        return list(output.shape), list(sequence_mask)
+
+    for test_inputs, test_outputs, mask_outputs in [
+        [(10, 50, 50, 50, 1, 1, 1, 1, (1, 1), (1, 1), 50, 50), [20, 100, 50], []],
+        [(10, 50, 50, 50, 1, 1, 1, 1, (2, 1), (1, 1), 50, 50), [20, 50, 50], []],
+        [(10, 50, 50, 50, 1, 1, 1, 1, (1, 1), (2, 1), 50, 50), [20, 50, 50], []],
+        [(10, 50, 50, 50, 1, 1, 1, 1, (2, 1), (2, 1), 50, 50), [20, 25, 50], []],
+        [(10, 50, 50, 50, 2, 2, 2, 2, (1, 1), (1, 1), 100, 100), [20, 100, 100], []],
+        [(10, 50, 50, 50, 3, 3, 3, 3, (1, 1), (1, 1), 150, 150), [20, 100, 150], []],
+        [(10, 50, 50, 50, 4, 4, 4, 4, (1, 1), (1, 1), 200, 200), [20, 100, 200], []],
+        [(10, 50, 50, 50, 32, 32, 64, 64, (1, 1), (1, 1), 3200, 3200), [20, 100, 3200], []],
+        [(10, 50, 50, 50, 32, 32, 64, 64, (1, 1), (1, 1), 3200, 50), [20, 100, 50], []],
+        [(10, 50, 50, 60, 4, 4, 4, 4, (1, 2), (1, 2), None, None), [20, 100, 60], []],
+        [(10, 50, 50, 50, 3, 3, 3, 3, (2, 3), (1, 1), None, None), [20, 50, 48], []],
+        [(10, 50, 50, 70, 4, 4, 4, 4, (1, 2), (1, 2), None, None), [20, 100, 68], []],
+        [(10, 50, 50, 50, 32, 32, 64, 64, (4, 3), (1, 3), None, None), [20, 25, 320], []],
+    ]:
+        shape, seq_mask = get_output_shape(*test_inputs)
+        assert shape == test_outputs
+
+
+def test_conformer_vgg_layer_pool_frontend_v1():
+    torch.manual_seed(42)
+
+    def get_output_shape(
+        batch,
+        time,
+        time_padding,
+        features,
+        conv1_channels,
+        conv2_channels,
+        conv3_channels,
+        conv4_channels,
+        conv2_stride,
+        pool_red,
+        in_dim,
+        out_dim,
+    ):
+        data_input = torch.randn(batch, time, features)
+        data_input = torch.cat((data_input, torch.zeros(batch, time_padding, features)), dim=1)
+        data_input = torch.cat((data_input, torch.randn(batch, time + time_padding, features)), dim=0)
+
+        sequence_mask = torch.zeros(batch, time)
+        sequence_mask = torch.cat((sequence_mask, torch.ones(batch, time_padding)), dim=1)
+        sequence_mask = torch.cat((sequence_mask, torch.zeros(batch, time + time_padding)), dim=0)
```

Comment:
Btw, another thing here: seq mask should be a bool tensor, I guess, but it is not here.